docs-en/chapter_hashing/hash_collision.md

```diff
@@ -1,45 +1,45 @@
 
-1. Improve the data structure of the hash table, **allowing it to function normally in the event of a hash collision**.
-2. Only perform resizing when necessary, i.e., when hash collisions are severe.
+1. Improve the hash table data structure so that, **the hash table can work normally when hash collisions occur**.
+2. Only perform resizing when necessary, i.e., when hash collisions are relatively severe.
 
 There are mainly two methods for improving the structure of hash tables: "Separate Chaining" and "Open Addressing".
 
 ## Separate Chaining
 
-In the original hash table, each bucket can store only one key-value pair. "Separate chaining" transforms individual elements into a linked list, with key-value pairs as list nodes, storing all colliding key-value pairs in the same list. The figure below shows an example of a hash table with separate chaining.
+In the original hash table, each bucket can store only one key-value pair. "Separate chaining" converts a single element into a linked list, treating key-value pairs as list nodes, storing all colliding key-value pairs in the same linked list. The figure below shows an example of a hash table with separate chaining.
 
 ![Separate Chaining Hash Table](hash_collision.assets/hash_table_chaining.png)
 
 The operations of a hash table implemented with separate chaining have changed as follows:
 
-- **Querying Elements**: Input `key`, pass through the hash function to obtain the bucket index, access the head node of the list, then traverse the list and compare `key` to find the target key-value pair.
+- **Querying Elements**: Input `key`, obtain the bucket index through the hash function, then access the head node of the linked list, traverse the linked list, and compare `key` to find the target key-value pair.
 - **Adding Elements**: First access the list head node via the hash function, then add the node (key-value pair) to the list.
-- **Deleting Elements**: Access the list head based on the hash function's result, then traverse the list to find and remove the target node.
+- **Deleting Elements**: Access the head of the linked list based on the result of the hash function, then traverse the linked list to find the target node and delete it.
 
 Separate chaining has the following limitations:
 
 - **Increased Space Usage**: The linked list contains node pointers, which consume more memory space than arrays.
-- **Reduced Query Efficiency**: Due to the need for linear traversal of the list to find the corresponding element.
+- **Reduced Query Efficiency**: Because linear traversal of the linked list is required to find the corresponding element.
```

Comment:
um optional: "Because linear traversal of th" -> "This is because linear traversal"